File: prototype/rails/src/test/domain/reports/payout_summary_test.rb
require "test_helper"

module Domain
  module Reports
    class PayoutSummaryTest < ActiveSupport::TestCase
      test "it counts the payouts a run wrote" do
        assert_equal 2, PayoutSummary.of([Money.from_cents(9_000), Money.from_cents(4_500)]).count
      end

      test "it totals the amounts paid out" do
        summary = PayoutSummary.of([Money.from_cents(9_000), Money.from_cents(4_500)])

        assert_equal "$135.00", summary.total.format
      end

      test "a run that paid nobody comes to nothing" do
        summary = PayoutSummary.of([])

        assert_equal 0, summary.count
        assert_equal "$0.00", summary.total.format
      end
    end
  end
end

File: app/src/app/Domain/Reports/PayoutSummaryTest.php
<?php

declare(strict_types=1);

namespace App\Domain\Reports;

use App\Domain\Money\Money;

it('counts the payouts a run wrote', function (): void {
    $summary = PayoutSummary::of([Money::fromCents(9000), Money::fromCents(4500)]);

    expect($summary->count)->toBe(2);
});

it('totals the amounts paid out', function (): void {
    $summary = PayoutSummary::of([Money::fromCents(9000), Money::fromCents(4500)]);

    expect($summary->total->format())->toBe('$135.00');
});

it('totals zero for a run that paid nobody', function (): void {
    $summary = PayoutSummary::of([]);

    expect($summary->count)->toBe(0)
        ->and($summary->total->format())->toBe('$0.00');
});
